### PR TITLE
RUNTIME_INIT synthetic beans - improve docs and error message

### PR DESCRIPTION
**Pull request by mkouba**
- is related to #11337

### CODE REVIEW

File: docs/src/main/asciidoc/cdi-reference.adoc [hunk 1]
```diff
@@ -814,14 +814,34 @@ The extended `BeanConfigurator` accepts either a `io.quarkus.runtime.RuntimeValu
...
+   return SyntheticBeanBuildItem.configure(Foo.class).scope(Singleton.class)
+                .setRuntimeInit() <2>
                 .runtimeValue(recorder.createFoo())
                 .done();
 }
 ----
+<1> The recorder must be executed in the `ExecutionTime.RUNTIME_INIT` phase.
+<2> The bean instance is initialized during `RUNTIME_INIT`.
 
+IMPORTANT: Synthetic bean initialized during `RUNTIME_INIT` must not be accessed during `STATIC_INIT`. `RUNTIME_INIT` build steps that access a runtime-init synthetic bean should consume the `SyntheticBeansRuntimeInitBuildItem`.
```
Comment (geoand) [inline docs/src/main/asciidoc/cdi-reference.adoc:844]: Can we have an example, or a link to some code to make this more tangible?

Comment (mkouba) [reply docs/src/main/asciidoc/cdi-reference.adoc:844]: Hm, I don't think we use it anywhere ATM. I will add an example...

**Review by geoand — APPROVED** [review]
Nice clarification!

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Thanks @mkouba !

**Pull request merged**